WalletWasabi/BitcoinCore/Processes/BitcoindRpcProcessBridge.cs

```diff
@@ -138,18 +139,27 @@ public async Task StopAsync(bool onlyOwned)
 
 				try
 				{
+					bool isKilled = false;
+
 					try
 					{
+						// Stop Bitcoin daemon using RPC "stop" command.
+						// The command actually only initiates the bitcoind graceful shutdown procedure.
+						// Our time budget for the bitcoind to stop is given by "ReasonableCoreShutdownTimeout".
 						await RpcClient.StopAsync().ConfigureAwait(false);
 					}
 					catch (Exception ex)
 					{
 						Logger.LogWarning(ex);
 						process.Kill();
```

Comment:
`WaitForExitAsync` has a killoncancel mechanism, worth to take a look maybe it can be used here?

Reply:
If RPC command fails for some reason, we don't really have too many options than to kill the process explicitly but the killing in `WaitForExitAsync` is tied to the cancelling mechanism, so I would use it in a non-standard way. 

I can remove `isKilled` variable because `Process.HasExited` in `WaitForExitAsync` handles that case but are we trying to make the code as short as possible or rather clear. In this case it looks weird to me to kill a process and then wait for it to exit.